Ginger/GingerCore/Drivers/Appium/SeleniumAppiumDriver.cs

```diff
@@ -201,78 +224,103 @@ public bool ConnectToAppium()
+                {
                     DriverDeviceType = eDeviceType.Tablet;
+                }
                 else
+                {
                     DriverDeviceType = eDeviceType.Phone;
+                }
 
-                //Setting capabilities
-                DriverOptions driverOptions = this.GetCapabilities();
-
-                driverOptions.AddAdditionalCapability("newCommandTimeout", INIT_TIMEOUT_SEC.TotalSeconds);
-
-                foreach (DriverConfigParam UserCapability in AdvanceDriverConfigurations)
+                //set timeout
+                if (NewCommandTimeout < 0)
                 {
-                 
-                    driverOptions.AddAdditionalCapability(UserCapability.Parameter, UserCapability.Value);
+                    NewCommandTimeout = 120;
                 }
+                TimeSpan commandTimeoutAsTimeSpan = TimeSpan.FromSeconds(NewCommandTimeout);
+                
+                //Setting capabilities                                
+                DriverOptions driverOptions = this.GetCapabilities();
+
                 //creating driver
                 switch (DriverPlatformType)
                 {
                     case eSeleniumPlatformType.Android:
-                        Driver = new AndroidDriver<AppiumWebElement>(serverUri, driverOptions, INIT_TIMEOUT_SEC);
+                        Driver = new AndroidDriver<AppiumWebElement>(serverUri, driverOptions, commandTimeoutAsTimeSpan);
                         break;
                     case eSeleniumPlatformType.iOS:
-                        Driver = new IOSDriver<AppiumWebElement>(serverUri, driverOptions, INIT_TIMEOUT_SEC);
+                        Driver = new IOSDriver<AppiumWebElement>(serverUri, driverOptions, commandTimeoutAsTimeSpan);
                         break;
                     case eSeleniumPlatformType.AndroidBrowser:
-                        Driver = new AndroidDriver<AppiumWebElement>(serverUri, driverOptions, INIT_TIMEOUT_SEC);
+                        Driver = new AndroidDriver<AppiumWebElement>(serverUri, driverOptions, commandTimeoutAsTimeSpan);
                         Driver.Navigate().GoToUrl("http://www.google.com");
                         break;
                     case eSeleniumPlatformType.iOSBrowser:
                         //TODO: start ios-web-proxy automatically
-                        Driver = new IOSDriver<AppiumWebElement>(serverUri, driverOptions, INIT_TIMEOUT_SEC);
+                        Driver = new IOSDriver<AppiumWebElement>(serverUri, driverOptions, commandTimeoutAsTimeSpan);
                         break;
                 }
                 mSeleniumDriver = new SeleniumDriver(Driver); //used for running regular Selenium actions
-                if (ImplicitWait == 0)
-                    ImplicitWait = 10;
-              //  Driver.Manage().Timeouts().ImplicitWait=TimeSpan.FromSeconds((int)ImplicitWait);
+                //if (ImplicitWait == 0)
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6113827)